app/models/manageiq/providers/vmware/infra_manager/provision/customization.rb

```diff
@@ -110,8 +110,11 @@ def collect_nic_settings
     nics = Array.wrap(options[:nic_settings])
     nic = nics[0]
     nic = {} if nic.blank?
-    [:dns_domain, :dns_servers, :sysprep_netbios_mode, :wins_servers, :addr_mode,
-     :gateway, :subnet_mask, :ip_addr].each { |key| nic[key] = options[key] }
```

Comment:
NOTE setting `nic[key] = options[key]` without checking if `nic[key]` is set already was causing the first entry in the `nic_settings` array to be overwritten with all `nil`s